index.bs

```diff
@@ -292,15 +306,134 @@ the PrivacyCG.
+constant which is a [=duration=] representing how long user activations will
+protect a [=host=] from storage deletion.
+
+The user agent defines a <dfn>bounce tracking timer period</dfn> constant which
+is a [=duration=] representing how often to run its timer algorithm.
+
+<p class=note>
+Schemeless site is used as the data structure key because by default cookies
+are sent to both `http://` and `https://` pages on the same domain.
+</p>
 
 <h3 id="bounce-tracking-mitigations-algorithms">Algorithms</h3>
 
 * TODO: Define the steps necessary to detect and store a "bounce".
-* TODO: Define the steps necessary to store user interactions.
-* TODO: Define the steps to analyze information in the data model and delete
-        appropriate sites.
+
+<h4 id="bounce-tracking-mitigations-activation-monkey-patch">User Activation Monkey Patch</h4>
+
+<div algorithm>
+
+To <dfn>record a user activation</dfn> given a [=Document=] |document|, perform
+the following steps:
+
+1. If |document|'s [=node navigable=] is null, then abort these steps.
+1. Let |settings| be |document|'s [=relevant settings object=].
+1. Let |origin| be |settings|' [=environment/top-level origin=].
+1. If |origin| is an [=opaque origin=] then abort these steps.
+1. Let |site| be the result of running [=obtain a site=] given |origin|.
+1. Set [=user activation map=][|site|'s [=host=]] to |settings|'s
+    [=environment settings object/current wall time=].
+
+</div>
+
+Append the following steps to the [=activation notification=] steps in the
+[[HTML#user-activation-processing-model|user activation processing model]]:
+
+1. Run [=record a user activation=] given <var ignore>document</var>.
+
+<h4 id="bounce-tracking-mitigations-timer">Timer</h4>
+
+<div algorithm>
+
+To run the <dfn>bounce tracking timer</dfn> algorithm given a [=moment=] |now|,
+perform the following steps:
+
+1. [=map/For each=] |host| -> |bounceTime| of [=candidate bounce tracking map=]:
+    1. If |bounceTime| + [=bounce tracking grace period=] is less than or equal to
+        |now|, then [=iteration/continue=].
+    1. Let |activationTime| be [=user activation map=][|host|].
+    1. If |activationTime| is not null and |activationTime| +
+        [=bounce tracking activation lifetime=] is greater than or equal to |now|,
+        then [=iteration/continue=].
+    1. If there is a [=top-level traversable=] whose
+        [=navigable/active document=]'s [=Document/origin=]'s
+        [=obtain a site|site=]'s [=host=] equals |host|,
+        then [=iteration/continue=].
+    1. [=Clear cookies for host=] given |host|.
+    1. [=Clear non-cookie storage for host=] given |host|.
+    1. [=Clear cache for host=] given |host|.
+
+<p class=note>TODO: Do something to prevent repeated deletions, etc.</p>
+
+</div>
+
+Every [=bounce tracking timer period=] the user agent should run the
+[=bounce tracking timer=] algorithm given the
+[=environment settings object/current wall time=].
```

Comment:
I'm changing this to just pass `[=wall clock=]'s [=unsafe current time=]`.  I don't think coarsening is really necessary since action will be performed at most once every timer period which is expected to be on the order of minutes or hours.  You can't really make a high precision clock out of this mechanism.